packages/common/src/core/schemas/internal/getLocationOptions.ts

```diff
@@ -22,9 +23,69 @@ export async function getLocationOptions(
   portalName: string,
   hubRequestOptions: IHubRequestOptions
 ): Promise<IHubLocationOption[]> {
-  const defaultExtent = await getGeographicOrgExtent(hubRequestOptions);
+  const typeFromEntity: any = getTypeFromEntity(entity);
+
+  switch (typeFromEntity) {
+    case "content":
+      return getContentLocationOptions(entity, hubRequestOptions);
+    default:
+      return getDefaultLocationOptions(entity, portalName, hubRequestOptions);
+  }
+}
+
+// TODO: Refactor parameters, they are icky gross
+// TODO: Maybe move these to outside of core and into respective entities
+
+export async function getContentLocationOptions(
+  entity: ConfigurableEntity,
+  hubRequestOptions: IHubRequestOptions
+): Promise<IHubLocationOption[]> {
+  const defaultExtent: IExtent = await getGeographicOrgExtent(
+    hubRequestOptions
+  );
   const location: IHubLocation = entity.location;
+  return (
+    [
+      {
+        label: "{{shared.fields.location.none:translate}}",
+        location: { type: "none" },
+      },
+      {
+        label: "{{shared.fields.location.itemExtent:translate}}",
+        entityType: "content",
+        location: {
+          type: "custom",
+          // TODO: Add custom bbox option here
+          // TODO: Remove "Add another location?" notification that appears when selecting a location
+          extent: extentToBBox(defaultExtent),
+          spatialReference: defaultExtent.spatialReference,
+        },
+      },
+    ] as IHubLocationOption[]
+  ).map((option) => {
+    // If this is a new entity, select the custom option by default
+    if (!entity.id && option.location.type === "custom") {
+      option.selected = true;
+    } else if (entity.id && !location && option.location.type === "none") {
+      option.selected = true;
+    } else if (location?.type === option.location.type) {
+      option.location = location;
+      option.selected = true;
+    }
```

Comment:
It took me a while to figure out why you had these if statements. I think my biggest hangup is the dual purpose of checking for the option's type AND seeing if a fallback is necessary. The re-assigning of `option.location` also is oddly placed, since the `custom` option is the only one with a location to begin with.

Don't take too long, but take a crack at simplifying this. Try to break up these 3 steps:
- Determining which location type is selected, including fallbacks
- Setting the `custom` option's `location` field
- Actually marking an option as selected

Reply:
I agree that coupling those operations do make this a little hard to follow, but there is a bigger issue that needs to be resolved first, which is that we need to base the options and selected state on the item's extent and `properties.boundary` and **not**  the org's extent and `properties.location`.